File: src/abc437b.cpp
#include<bits/stdc++.h>
using namespace std;
#define rep(i,n) for(int i=0;i<(n);i++)
int main(){
  int h,w,n;cin>>h>>w>>n;
  vector<vector<int>>a(h,vector<int>(w));
  rep(i,h)rep(j,w)cin>>a[i][j];
  vector<int>b(n);rep(i,n)cin>>b[i];
  int ans=0;
  rep(i,h){
    int cnt=0;
    rep(j,n)cnt+=count(a[i].begin(),a[i].end(),b[j]);
    ans=max(ans,cnt);
  }
  cout<<ans<<'\n';
}

File: src/abc437b.py
import sys
data = list(map(int, sys.stdin.read().split()))
ptr = 0

h = data[ptr]; ptr += 1
w = data[ptr]; ptr += 1
n = data[ptr]; ptr += 1
a = []
for i in range(h):
  a.append(data[ptr:ptr + w])
  ptr += w
b = data[ptr:ptr + n]

# print(a)
# print(b)
ans = 0
for row in a:
  cnt = 0
  for x in row:
    cnt += b.count(x)
  ans = max(cnt, ans)
print(ans)
